version: '3.8'
services:

  db:
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
    image: postgres:13.0-alpine
    volumes:
      - /var/lib/postgres/data/
    env_file:
      - ./.env
  web:
    image: zyrker/backend_foodgram:v1
    restart: always
    command: >
      sh -c "
            python manage.py collectstatic --noinput &&
            python manage.py migrate &&
            python manage.py add_data_from_csv &&
            gunicorn foodgram.wsgi:application --bind 0:8000"
    volumes:
      - static_value:/app/static/
      - media_value:/app/media/
    depends_on:
      - db
    env_file:
      - ./.env
  frontend:
    image: zyrker/frontend_foodgram:v1
    depends_on:
      - web
    volumes:
      - ../frontend/:/app/result_build/

  nginx:
    image: nginx:1.19.3
    ports:
      - "80:80"
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ../frontend/build:/usr/share/nginx/html/
      - ../docs/:/usr/share/nginx/html/api/docs/
      - static_value:/var/html/static/
      - media_value:/var/html/media/
    depends_on:
      - frontend

volumes:
  static_value:
  media_value:
  postgres_data:
